TSG.md

```diff
@@ -0,0 +1,207 @@
+
+Solution: Do an 'ls -l /etc/cron.daily/mlocate' at the shell prompt. If "mlocate" is added to the /etc/cron.daily then Blobfuse2 must be whitelisted, so that the Blobfuse2 mount directory is not scanned by updatedb. This is done by updating the updatedb.conf file .
+cat /etc/updatedb.conf
+It should look like this.
+PRUNE_BIND_MOUNTS="yes"
+
+PRUNENAMES=".git .bzr .hg .svn"
+
+PRUNEPATHS="/tmp /var/spool /media /var/lib/os-prober /var/lib/ceph /home/.ecryptfs /var/lib/schroot"
+
+PRUNEFS="NFS nfs nfs4 rpc_pipefs afs binfmt_misc proc smbfs autofs iso9660 ncpfs coda devpts ftpfs devfs devtmpfs fuse.mfs shfs sysfs cifs lustre tmpfs usbfs udf fuse.glusterfs fuse.sshfs curlftpfs ceph fuse.ceph fuse.rozofs ecryptfs fusesmb"
+
+1) Add the Blobfuse2 mount path eg: /mnt to the PRUNEPATHS
+OR
+
+1) Add "Blobfuse2" and "fuse" to the PRUNEFS
+
+It won't harm to do both.
+
+Below are the steps to automate this at pod creation:
+
+1.Create a new configmap in the cluster which contains the new configuration about the script.
+
+2.Create a DaemonSet with the new configmap which could apply the configuration changes to every node in the cluster.
+```
+Example:
+configmap fiie: (testcm.yaml)
+apiVersion: v1
+kind: ConfigMap
+metadata:
+name: testcm
+data:
+updatedb.conf: |
+PRUNE_BIND_MOUNTS="yes"
+PRUNEPATHS="/tmp /var/spool /media /var/lib/os-prober /var/lib/ceph /home/.ecryptfs /var/lib/schroot /mnt /var/lib/kubelet"
+PRUNEFS="NFS nfs nfs4 rpc_pipefs afs binfmt_misc proc smbfs autofs iso9660 ncpfs coda devpts ftpfs devfs devtmpfs fuse.mfs shfs sysfs cifs lustre tmpfs usbfs udf fuse.glusterfs fuse.sshfs curlftpfs ceph fuse.ceph fuse.rozofs ecryptfs fusesmb fuse Blobfuse2"
+DaemonSet file: (testcmds.yaml)
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+name: testcmds
+labels:
+test: testcmds
+spec:
+selector:
+matchLabels:
+name: testcmds
+template:
+metadata:
+labels:
+name: testcmds
+spec:
+tolerations:
+- key: "kubernetes.azure.com/scalesetpriority"
+operator: "Equal"
+value: "spot"
+effect: "NoSchedule"
+containers:
+- name: mypod
+image: debian
+volumeMounts:
+- name: updatedbconf
+mountPath: "/tmp"
+- name: source
+mountPath: "/etc"
+command: ["/bin/bash","-c","cp /tmp/updatedb.conf /etc/updatedb.conf;while true; do sleep 30; done;"]
+restartPolicy: Always
+volumes:
+- name: updatedbconf
+configMap:
+name: testcm
+items:
+- key: "updatedb.conf"
+path: "updatedb.conf"
+- name: source
+hostPath:
+path: /etc
+type: Directory
+```
+**4. File contents are not in sync with storage** 
+
+Please refer to the file cache component setting `timeout-sec`.
+
+
+**5. failed to unmount /path/<mount dir>**
+
+Unmount fails when a file is open or a user or process is cd'd into the mount directory or its sub directories. Please ensure no files are in use and try the unmount command again. Even umount -f will not work if the mounted files /directories are in use.
```

Comment:
Sometimes customer complain that there temp cache directory is not wiped out. We have observed that they do not unmount blobfuse gracefully and directly terminate the vm or container. So in such cases that directory will remain in place. We can mention here that always unmount before you kill the vm/container. And if such cases are possible then use the flags in v2 that cleans up the temp directory on start.